kernel/src/hil/acifc.rs

```diff
@@ -0,0 +1,40 @@
+//! Interface for direct control of ACIFC
+//!
+//! Author: Danilo Verhaert <verhaert@cs.stanford.edu>
+
+use returncode::ReturnCode;
+
+/// Enum for selecting which edge to trigger interrupts on.
+// #[derive(Debug)]
+// pub enum InterruptMode {
+//     RisingEdge,
+//     FallingEdge,
+//     EitherEdge,
+// }
+
+pub trait Acifc {
```

Comment:
High-level, this HIL needs to be fleshed out significantly. To begin with, it would be good to have much more comments. For one, I think trying to actually explain in prose what the purpose of the HIL is (the kind of hardware it abstracts), as well as what users can expect and what implementors should guarantee would help draw out issues with the interface and second, we just need documentation :)

Reply:
I am always a fan of more documentation. I'll add way more.

Reply:
What should be the central location of the comments? In sam4l, capsules or the HIL?